### PR TITLE
test: add restrictable vars tests

### PR DESCRIPTION
**Pull request by mlutze**
related to #5229

### CODE REVIEW

**Comment by JonathanStarup** [comment]
Based on #5311

File: main/src/ca/uwaterloo/flix/language/phase/unification/CaseSetUnification.scala [hunk 1]
```diff
@@ -49,25 +50,35 @@ object CaseSetUnification {
...
 
-//        tpe1 match {
-//          case x: Type.Var if renv.isFlexible(x.sym) =>
-//            if (tpe2 eq Type.All)
-//              return Ok(Substitution.singleton(x.sym, Type.All))
-//            if (tpe2 eq Type.Empty)
-//              return Ok(Substitution.singleton(x.sym, Type.Empty))
-//
-//          case _ => // nop
+    // TODO RESTR-VARS this doesn't help lol
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/unification/CaseSetUnification.scala:53]: It does not help? :(

Comment (mlutze) [reply main/src/ca/uwaterloo/flix/language/phase/unification/CaseSetUnification.scala:53]: Makes it worse somehow. But this was a naive translation from the other file. Probably something smarter can be done here.

File: main/src/ca/uwaterloo/flix/language/phase/inference/RestrictableChooseInference.scala [hunk 2]
```diff
@@ -216,12 +216,13 @@ object RestrictableChooseInference {
...
         )
-        _ <- unifySubset(indexOutVar, set, enumSym, loc)
+//        _ <- unifySubset(indexOutVar, set, enumSym, loc)
+        _ <- unifySubset(set, indexOutVar, enumSym, loc)
 
         resultTconstrs = constrs ::: constrss.flatten
 
         // τ_out
-        resultTpe <- unifyTypeM(tpe0 :: tpes, loc)
+        resultTpe <- unifyTypeM(enumTypeOut, tpe0, loc)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/inference/RestrictableChooseInference.scala:225]: And this too I guess?

Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/inference/RestrictableChooseInference.scala:225]: No this was just a bug bug that made the unification and reconstruction inconsistent

File: main/src/ca/uwaterloo/flix/language/phase/inference/RestrictableChooseInference.scala [hunk 3]
```diff
@@ -216,12 +216,13 @@ object RestrictableChooseInference {
           enumSym,
           loc
         )
-        _ <- unifySubset(indexOutVar, set, enumSym, loc)
+//        _ <- unifySubset(indexOutVar, set, enumSym, loc)
+        _ <- unifySubset(set, indexOutVar, enumSym, loc)
```
[the remark this answers is not in the document]
Comment (mlutze) [reply main/src/ca/uwaterloo/flix/language/phase/inference/RestrictableChooseInference.scala]: https://github.com/flix/flix/pull/5317

File: main/src/ca/uwaterloo/flix/language/phase/unification/CaseSetUnification.scala [hunk 4]
```diff
@@ -84,7 +84,9 @@ object CaseSetUnification {
     /// Run the expensive boolean unification algorithm.
     ///
     implicit val universe: Universe = Universe(cases, enumSym)
-    booleanUnification(eraseAliases(tpe1), eraseAliases(tpe2), renv)
+    val input1 = TypeMinimization.minimizeType(simplify(eraseAliases(tpe1)))
+    val input2 = TypeMinimization.minimizeType(simplify(eraseAliases(tpe2)))
+    booleanUnification(input1, input2, renv)
```
Comment (JonathanStarup) [inline main/src/ca/uwaterloo/flix/language/phase/unification/CaseSetUnification.scala:89]: This is the only non-testing changes, it enables some of the tests to run without blowing up so we thought it valid to keep in this PR.

**Comment by JonathanStarup** [comment]
@magnus-madsen Ready for review/merge

**Comment by magnus-madsen** [comment]
What about the use cases from the paper? Can we have those too? (I think they are in a separate PR).

**Pull request merged**